simulator/modules/decode/bypass/data_bypass_interface.h

```diff
@@ -9,40 +9,55 @@
 #define DATA_BYPASS_INTERFACE_H
 
 #include <infra/types.h>
+#include <infra/ports/timing.h>
 #include <infra/macro.h>
 
 class RegisterStage
 {
 public:
-    constexpr explicit RegisterStage(uint8 value) : value(value) { }
-
     auto operator==(const RegisterStage& rhs) const { return value == rhs.value; }
-    explicit operator uint8() const { return value; }
 
     void inc() { ++value; }
 
-    static constexpr const uint8 BYPASSING_STAGES_NUMBER = 3;
-    static constexpr RegisterStage in_RF() { return RegisterStage(IN_RF_STAGE_VALUE); }
+    static constexpr const uint8 BYPASSING_STAGES_NUMBER = 4;
+
+    static void set_complex_arithmetic_latency_value( uint8 value) 
+    { 
+        LAST_EXECUTION_STAGE_VALUE = value - 1;
+    }
+
+    // returns the latency between the first and the last execution stages
+    static auto get_last_execution_stage_latency()
```

Comment:
`static` is a bad design here (and everywhere), it will cost some pain to fix if we go to multicluster/multicore modeling  and/or unit testing,
If possible, put it is a non-static variable to DataBypass class, defining it in the ctor.